Add Path assertions to WithAssertions

cykl opened the pull request:
#### Check List:
* Unit tests : YES
* Javadoc with a code example (API only) : NA

Hi Joel,

I just noticed that Path assertions are absent from the `WithAssertions` interface. Is it an inadvertent omission or a deliberate omission ?

joel-costigliola commented:
I would go for inadvertent ;-) 
We have a test to check that `Assertions` and `BDDAssertions` are in sync, we should use the same strategy for `WithAssertions`.

cykl commented:
I can give it a look if you like. Maybe another PR, though.

cykl commented:
A quick test showed a lot of discrepancies. I will try to fix them and update this PR or fill a new one.

joel-costigliola commented:
you can update this PR, don't bother creating a new one.

filiphr commented:
@cykl there are 12 `assertThat` methods which are missing. 

@joel-costigliola, should we use the same strategy as with the `SoftAssertions` and the `BddAssertions`. Shouldn't we check for the other methods? Shouldn't `WithAssertions` be exactly like the `Assertions`?

cykl commented:
@filiphr I added a test to `AssertionsTest`  and fixed all the issues. The only remaining discrepancies are Iterable and Iterator. Methods were present in both classes but with different return types. Surprisingly, compiler got confused when I replaced  `AbstractIterableAssert` by `IterableAssert`, but I haven' t yet investigated the issue. Is this a known issue ?

Today was a busy day, I will push the patch over the weekend.

filiphr commented:
@cykl That is strange. There shouldn't be an issue in that. The same thing is working on the `Assertions`. I also just tried it locally on my side and there are no problems.

joel-costigliola commented:
Integrated thanks Clément.
